### ISSUE TITLE
[BUG] Inconsistent product dims

### ISSUE THREAD

**Issue by rzinke** [aria-tools/ARIA-tools#364]
**Describe the bug**
I recently ordered some new (reprocessed) data for Tibet track 41. When I attempt to process those data for timeseries analysis using ARIA-tools, however, I get the error `Exception(f'Inconsistent product dims ...`. I understand that this is by design, as a check for consistent product dims was recently added; however, I'm not sure what to do about it.

Any suggestions?

***Full error output***
```
1164 out of 2936 interferograms rejected for not meeting specified spatial thresholds
Download/cropping DEM
Applied cutline to produce 3 arc-sec SRTM DEM: 9arc-no_v4/DEM/glo_90.dem
***Downloading water mask... ***

Extracting unwrapped phase, coherence, and connected components for each interferogram pair
Generating: unwrappedPhase - [>                                                 ]Traceback (most recent call last):
  File "/home/rzinke/miniconda3/envs/DevARIA-tools/bin/ariaTSsetup.py", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/home/rzinke/tools/ARIA-tools/tools/bin/ariaTSsetup.py", line 18, in <module>
    main(inps)
  File "/home/rzinke/tools/ARIA-tools/tools/ARIAtools/tsSetup.py", line 474, in main
    arrshape = export_products(standardproduct_info.products[1],
  File "/home/rzinke/tools/ARIA-tools/tools/ARIAtools/extractProduct.py", line 1164, in export_products
    raise Exception(f'Inconsistent product dims between'
Exception: Inconsistent product dims betweenproducts {outname} and {prev_outname}:check respective width ({ref_wid}, {ref_height})and height ({prod_wid}, {prod_height})
```

**Desktop (please complete the following information):**
 - ARIA-tools git tag: commit 54165442eddb3fe2824a08a30a0810ca9144e1c3 (HEAD -> sss_mlbugfix, origin/sss_mlbugfix)
 - OS: Linux CentOS

**Comment by sssangha** [comment]
Hello @rzinke I have not encountered this error.

With that said, I've introduced a [PR](https://github.com/aria-tools/ARIA-tools/pull/363) with more robust checks + better handling of product resolution.

Could you please test it out for your case, and if this error persists could you perhaps provide a run example over a small stack I can debug? Thanks!

**Comment by rzinke** [comment]
Hi @sssangha, I'm getting a similar error when I run `ariaTSsetup.py` for a different track (track 085). ARIA-tools seems to stitch the unwrapped phase, incidenceAngle, lookAngle, and azimuthAngle just fine, but then throws an error during the coherence step.

The error is:
```
Extracting single incidence angle, look angle and azimuth angle files valid over common interferometric grid
Generating: incidenceAngle - [==================================================] 20141210_20141116
Generating: lookAngle - [==================================================] 20141210_20141116
Generating: azimuthAngle - [==================================================] 20141210_20141116
Traceback (most recent call last):
  File "/home/rzinke/miniconda3/envs/sssARIA-tools/bin/ariaTSsetup.py", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/home/rzinke/tools/ARIA-tools/tools/bin/ariaTSsetup.py", line 18, in <module>
    main(inps)
  File "/home/rzinke/tools/ARIA-tools/tools/ARIAtools/tsSetup.py", line 497, in main
    dim_check(ref_arr_record, prod_arr_record)
  File "/home/rzinke/tools/ARIA-tools/tools/ARIAtools/vrtmanager.py", line 569, in dim_check
    raise Exception('Inconsistent product dims '
Exception: Inconsistent product dims between products /data/Tibet/Track085/Timeseries/aoiTianShan/9arc/azimuthAngle/20141210_20141116 and /data/Tibet/Track085/Timeseries/aoiTianShan/9arc/coherence/20221005_20220923: check respective width (None, None) and height (1738, 1739) and geotrans ((79.819968072, 0.002499999, 0.0, 46.219981512, 0.0, -0.002499999), (79.8196564112118, 0.00249951952186421, 0.0, 46.2184349894632, 0.0, -0.00249964494264268))
```

I'm working with 
commit 72680c6f5b853bdffc73e718db48fd9b633b342b (HEAD -> sss_metafix, origin/sss_metafix)

**Comment by rzinke** [comment]
I overlaid the coherence map for 20221005_20220923 on top of the unwrapped phase map. I can see the pixels of the coherence (viridis) don't fully cover the unw phase (gray).

<img width="785" alt="Screenshot 2023-08-03 at 8 07 44 AM" src="https://github.com/aria-tools/ARIA-tools/assets/48299159/498955b0-f8d7-481f-aa5c-350b59e41993">

**Comment by sssangha** [comment]
Hello @rzinke , thanks for bringing this to my attention.

I was able to work out a patch. Please refer to PR #377 linked above.

Note I'd advise to first run a minimum working example sanity check in the interest of time. E.g. I ran a test using the following skinny subset of products:
`S1-GUNW-A-R-085-tops-20141210_20141116-122205-00084E_00028N-PP-98e6-v2_0_5.nc
S1-GUNW-A-R-085-tops-20141210_20141116-122232-00083E_00029N-PP-489c-v2_0_5.nc
S1-GUNW-A-R-085-tops-20141210_20141116-122256-00083E_00031N-PP-0189-v2_0_5.nc
S1-GUNW-A-R-085-tops-20141210_20141116-122321-00083E_00033N-PP-e2d9-v2_0_5.nc
S1-GUNW-A-R-085-tops-20141210_20141116-122346-00082E_00034N-PP-84a1-v2_0_5.nc
S1-GUNW-A-R-085-tops-20141210_20141116-122411-00082E_00035N-PP-ae56-v2_0_5.nc
S1-GUNW-A-R-085-tops-20141210_20141116-122437-00082E_00037N-PP-b5eb-v2_0_5.nc
S1-GUNW-A-R-085-tops-20141210_20141116-122502-00081E_00039N-PP-10a6-v2_0_5.nc
S1-GUNW-A-R-085-tops-20141210_20141116-122526-00081E_00040N-PP-9a4e-v2_0_5.nc
S1-GUNW-A-R-085-tops-20141210_20141116-122551-00081E_00042N-PP-29d3-v2_0_5.nc
S1-GUNW-A-R-085-tops-20141210_20141116-122616-00080E_00043N-PP-1ecc-v2_0_5.nc
S1-GUNW-A-R-085-tops-20141210_20141116-122644-00080E_00044N-PP-a60b-v2_0_5.nc
S1-GUNW-A-R-085-tops-20221005_20220923-122248-00084E_00027N-PP-1159-v2_0_5.nc
S1-GUNW-A-R-085-tops-20221005_20220923-122312-00084E_00029N-PP-8d31-v2_0_5.nc
S1-GUNW-A-R-085-tops-20221005_20220923-122337-00083E_00030N-PP-1243-v2_0_5.nc
S1-GUNW-A-R-085-tops-20221005_20220923-122402-00083E_00032N-PP-b0f9-v2_0_5.nc
S1-GUNW-A-R-085-tops-20221005_20220923-122427-00083E_00033N-PP-429f-v2_0_5.nc
S1-GUNW-A-R-085-tops-20221005_20220923-122452-00082E_00035N-PP-957e-v2_0_5.nc
S1-GUNW-A-R-085-tops-20221005_20220923-122517-00082E_00036N-PP-12fe-v2_0_5.nc
S1-GUNW-A-R-085-tops-20221005_20220923-122542-00081E_00038N-PP-d754-v2_0_5.nc
S1-GUNW-A-R-085-tops-20221005_20220923-122607-00081E_00039N-PP-54bb-v2_0_5.nc
S1-GUNW-A-R-085-tops-20221005_20220923-122632-00081E_00041N-PP-c4a9-v2_0_5.nc
S1-GUNW-A-R-085-tops-20221005_20220923-122657-00080E_00042N-PP-e870-v2_0_5.nc`

And I used this command:
`ariaTSsetup.py -f 'products/*.nc' -w test_TS -of ISCE -d download -m Download -nt 8 -ml 3 -v`


Please give it a go and let me know if this resolves the issue. Thanks

**Comment by sssangha** [comment]
Fixed in PR #377
